dataactvalidator/app.py

```diff
@@ -149,14 +97,126 @@ def run_app():
                         pass
 
 
-def get_current_job():
-    """Return the job currently stored in flask.g"""
-    # The job_id is added to flask.g at the beginning of the validate route. We expect it to be here now, since
-    # validate is currently the app's only functional route
-    job_id = g.get('job_id', None)
-    if job_id:
-        sess = GlobalDB.db().session
-        return sess.query(Job).filter(Job.job_id == job_id).one_or_none()
+def validator_process_file_generation(sess, file_gen_id, is_local):
+    """ Retrieves a FileGeneration object based on its ID, and kicks off a file generation. Handles errors by ensuring
+        the FileGeneration (if exists) is no longer cached.
+
+    Args:
+        sess: Current database session
+        file_gen_id: ID of a FileGeneration object
+        is_local: A boolean flag indicating whether the application is being run locally or not
+
+    Raises:
+        Any Exceptions raised by the FileGenerationManager
+    """
+    file_generation = None
+
+    try:
+        file_generation = sess.query(FileGeneration).filter_by(file_generation_id=file_gen_id).one_or_none()
+        if file_generation is None:
```

Comment:
Shouldn't a queue not dispense the same message to two different instances making requests, isn't that the point of a queue? Also, the only way for `file_generation` to be `None` is if a `FileGeneration` object has never existed with that `ID`. It's not making a query for an incomplete `FileGeneration`, just a `FileGeneration` all together.

Reply:
> Shouldn't a queue not dispense the same message to two different instances making requests, isn't that the point of a queue?
yes, typically that's the default setup. But depends on the transaction handling of the issue, and things like the `VisibilityTimeout`, and the _exactly-once_ or _at-most-once_ configuration design of the queue. I just not sure what these are in our scenario.

Most always a queue will only allow 1 consumer to process 1 message at a time. Some transactional or "rollback" behavior of a queues will often _leave_ a message in there while it's being handled, and make it invisible for a period of time, until it receives some ACK that the handling code completed. But if it doesn't get that "ACK", and the visibility timeout expires, it will show back up as being consumable. In which case you _could_ have 2 things processing the same message. Or if something processing it fails, the message may be put back in the queue (which may cause a poison message to block the queue) or eventually drop into a dead-letter-queue.  

RE: your 2nd comment. So we don't delete processed `FileGeneration` objects? If so, they we should be good ... it would just find the _completed_ job.

Reply:
All this ^^ boils down to whether you think, when a `file_generation` record is not found, should we 
a) log this as a warning, and continue (with the assumption that it was already generated or cancelled or something)
b) throw an error

I think you understand the context best, so your call.